src/Symfony/Component/Routing/Loader/ObjectLoader.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+use Symfony\Component\Config\Loader\Loader;
+use Symfony\Component\Config\Resource\FileResource;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * A route loader that calls a method on an object to load the routes.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+abstract class ObjectLoader extends Loader
+{
+    /**
+     * @deprecated since Symfony 4.4 and will be removed in 5.0, implement and use "getObject" instead.
+     */
+    abstract protected function getServiceObject($id);
```

Comment:
Are we allowed to remove the "abstract" part and throw a `\RuntimeException` to allow to not implement this method anymore?

Reply:
yes. Turning an abstract method into a concrete one is not a BC break, as long as the signature stays the same and the method is not made final.